### ISSUE TITLE
Duplicated AI goals and facets

### ISSUE THREAD

**Issue by virtualghetto** [wesnoth/wesnoth#4887]
### Game and System Information
Wesnoth 1.14.9
Debian sid package.

### Describe the bug
AI blocks defined inside `[ai]` appear to get duplicated in the game inspector.

Details are in this thread
https://forums.wesnoth.org/viewtopic.php?f=21&t=51764

**Comment by mattsc** [comment]
I finally figured out where this is coming from.  The tags get duplicated in [this part of the connect_engine.cpp code](https://github.com/wesnoth/wesnoth/blob/1.14/src/game_initialization/connect_engine.cpp#L1188).  That merge, in combination with the adding an `[ai]ai_algorithm=` tag at line 1078 turns this config:
```
[ai]
	ai_algorithm =
[/ai]
[ai]
	[goal]
		value = 5
		[criteria]
			side = 3
		[/criteria]
	[/goal]
[/ai]
```
into this one:
```
[ai]
	ai_algorithm =
	[goal]
		value = 5
		[criteria]
			side = 3
		[/criteria]
	[/goal]
[/ai]
[ai]
	[goal]
		value = 5
		[criteria]
			side = 3
		[/criteria]
	[/goal]
[/ai]
```

@gfgtdf In 1.15 you changed this part of the code in e036cadf0bb.  Unfortunately, the last sentence of the commit message is incomplete, so I don't know what in particular this also fixes w.r.t. subtags, but it would certainly fix this issue.  [I did verify that the problem does not exist in 1.15.]  However, I have no idea what has changed in this code between 1.14 and 1.15, so I'd prefer you to make that call.

**Comment by gfgtdf** [comment]
Hmm see the previous  discussion of this bug https://github.com/wesnoth/wesnoth/issues/3338  . But im not that much into this code so i cannot tell you more than what you see in that bugs discussion

**Comment by mattsc** [comment]
There's some discussion at the end of #3338 about backporting this, but apparently it never happened.  I'll backport that after the "emergency" release of 1.14.13 this week.  I don't want to do so without time for testing.

**Comment by mattsc** [comment]
This is now fixed in 1.14 also in 16022d571b.
